nixos/tests/mpdscheduler.nix

```diff
@@ -0,0 +1,138 @@
+import ./make-test-python.nix ({ pkgs, ... } :
+let
+  track = pkgs.fetchurl {
+    # Sourced from http://freemusicarchive.org/music/Blue_Wave_Theory/Surf_Music_Month_Challenge/Skyhawk_Beach_fade_in
+    # License: http://creativecommons.org/licenses/by-sa/4.0/
+
+    name = "Blue_Wave_Theory-Skyhawk_Beach.mp3";
+    url = "https://files.freemusicarchive.org/storage-freemusicarchive-org/music/ccCommunity/Blue_Wave_Theory/Surf_Music_Month_Challenge/Blue_Wave_Theory_-_04_-_Skyhawk_Beach.mp3";
+    sha256 = "0xw417bxkx4gqqy139bb21yldi37xx8xjfxrwaqa0gyw19dl6mgp";
+  };
+  musicService = {}: {
+    description = "Sets up the music file(s) for MPD to use.";
+    requires = [ "mpd.service" ];
+    after = [ "mpd.service" ];
+    wantedBy = [ "default.target" ];
+    script = ''
+      mkdir -p /var/lib/mpd/music && chown -R mpd:mpd /var/lib/mpd/music
+      cp ${track} /var/lib/mpd/music
+      chown mpd:mpd /var/lib/mpd/music/$(basename ${track})
+    '';
+  };
+in {
+  name = "mpdscheduler";
+  meta = with pkgs.stdenv.lib.maintainers; {
+    maintainers = [ aberDerBart ];
+  };
+
+  nodes = {
+    mpdWithSchedulerHost = { config, pkgs, ... }: {
+      boot.kernelModules = [ "snd-dummy" ];
+      sound.enable = true;
+      services.mpd = {
+        enable = true;
+        extraConfig = ''
+          audio_output {
+            type "alsa"
+            name "alsa"
+            mixer_type "null"
+          }
+
+          log_level "verbose"
+        '';
+      };
+      systemd.services.musicService = musicService {};
```

Comment:
You may be able to replace this with a single line, something more simple like...
```
systemd.tmpfiles.rules = [ "C /var/lib/music/Blue_Wave_Theory_-_04_-_Skyhawk_Beach.mp3 - - - - ${track}" ];
```

Untested, but should work.

Reply:
It didn't work:
```
systemd-tmpfiles[584]: Detected unsafe path transition /var/lib/mpd → /var/lib/mpd/music during canonicalization of /var/lib/mpd/music.
```
as `/var/lib/mpd/` belongs to user `mpd`, not `root`. 
However, 
```
systemd.tmpfiles.rules = [ "L /var/lib/music/Blue_Wave_Theory_-_04_-_Skyhawk_Beach.mp3 - - - - ${track}" ];
```
did (with setting `follow_outside_symlinks "yes"` in `mpd.extraConfig`)

Reply:
I specifically set the folder to something other than `/var/lib/mpd/music` to avoid the unsafe transition. If you want to use `/var/lib/mpd/music` you can set the owner to `mpd`, as you noticed:
```
systemd.tmpfiles.rules = [ "C /var/lib/music/Blue_Wave_Theory_-_04_-_Skyhawk_Beach.mp3 - mpd mpd - ${track}" ];
```